### PR TITLE
feat: Improve Glossary entries for symm/asymm crypto & ciphers (enc algorithms)

### PR DESCRIPTION
**Pull request by bsmth**
### Description

This PR improves some of the glossary entries relating to asymmetric and symmetric crypto. 

__Additions:__
* link to sublecrypto algos which has a table recommending different encryption algos per use
* Add details about elliptic-curve crypto (ecc). Likely ECDH and ECDSA are good to highlight.

__Removals:__
* `{{glossary("attack", "attacks")}}` doesn't exist, so removing this reference

### Motivation

A follow-up to https://github.com/mdn/content/pull/33393 where I spotted some more areas to improve.

### CODE REVIEW

File: files/en-us/glossary/cipher/index.md [hunk 1]
```diff
@@ -6,23 +6,24 @@ page-type: glossary-definition
...
+Modern ciphers are designed to withstand attacks discovered through {{glossary("cryptanalysis")}}. There is no guarantee that all attack methods have been discovered, so each algorithm is [recommended for different purposes](/en-US/docs/Web/API/SubtleCrypto#supported_algorithms) based on known classes of attacks.
 
-Ciphers operate two ways, either as [block ciphers](https://en.wikipedia.org/wiki/Block_cipher) on successive blocks, or buffers, of data, or as [stream ciphers](https://en.wikipedia.org/wiki/Stream_cipher) on a continuous data flow (often of sound or video).
+Ciphers operate either as [block ciphers](https://en.wikipedia.org/wiki/Block_cipher) on successive blocks (or buffers) of data, or as [stream ciphers](https://en.wikipedia.org/wiki/Stream_cipher) on a continuous data flow (often of sound or video).
 
-They also are classified according to how their {{glossary("key", "keys")}} are handled:
+Ciphers also are classified according to how their {{glossary("key", "keys")}} are handled:
```
Comment (PassionPenguin) [inline files/en-us/glossary/cipher/index.md:18]:
```suggestion
Ciphers are also classified according to how their {{glossary("key", "keys")}} are handled:
```

File: files/en-us/glossary/cipher/index.md [hunk 2]
```diff
@@ -6,23 +6,24 @@ page-type: glossary-definition
...
 
-- [symmetric key](https://en.wikipedia.org/wiki/Symmetric_key_algorithm) algorithms use the same key to encode and decode a message. The key also must be sent securely if the message is to stay confidential.
-- [asymmetric key](https://en.wikipedia.org/wiki/Asymmetric_key_algorithm) algorithms use a different key for encryption and decryption.
+- {{Glossary("Symmetric-key cryptography", "symmetric key")}} algorithms use the same key to encode and decode a message. The key also must be sent securely if the message is to stay confidential.
+- {{Glossary("Public-key cryptography", "asymmetric key")}} algorithms use one key for encryption and another for decryption.
```
Comment (PassionPenguin) [inline files/en-us/glossary/cipher/index.md:21]:
```suggestion
- {{Glossary("Public-key cryptography", "asymmetric key")}} algorithms use one key for encryption and the other for decryption.
```

**Review by PassionPenguin — COMMENTED** [review]
some suggestions :)

File: files/en-us/glossary/cipher/index.md [hunk 3]
```diff
@@ -6,23 +6,24 @@ page-type: glossary-definition
 
 {{GlossarySidebar}}
 
-In {{glossary("cryptography")}}, a **cipher** is an algorithm that can {{glossary("encryption", "encode")}} {{glossary("Plaintext")}} to make it unreadable, and to {{glossary("decryption", "decode")}} it back.
+In {{glossary("cryptography")}}, a **cipher** is an algorithm that can {{glossary("encryption", "encrypt")}} {{glossary("Plaintext")}} to make it unreadable, and to {{glossary("decryption", "decrypt")}} the encoded data back to plaintext again.
```
Comment (PassionPenguin) [inline files/en-us/glossary/cipher/index.md:9]: should we use the lowercase one `plaintext` in the macro here? 🤔

File: files/en-us/glossary/cipher/index.md [hunk 4]
```diff
@@ -6,23 +6,24 @@ page-type: glossary-definition
...
 
 Ciphers were common long before the information age (e.g., [substitution ciphers](https://en.wikipedia.org/wiki/Substitution_cipher), [transposition ciphers](https://en.wikipedia.org/wiki/Transposition_cipher), and [permutation ciphers](https://en.wikipedia.org/wiki/Permutation_cipher)), but none of them were cryptographically secure except for the [one-time pad](https://en.wikipedia.org/wiki/One-time_pad).
+[AES](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard), [RSA](<https://en.wikipedia.org/wiki/RSA_(cryptosystem)>) and [Blowfish](<https://en.wikipedia.org/wiki/Blowfish_(cipher)>) are examples of ciphers that are part of modern encryption standards and systems.
```
Comment (PassionPenguin) [inline files/en-us/glossary/cipher/index.md:12]: maybe we could add a linebreak between l11 & l12 here, and repharse it to make it more consistent like

```suggestion

In the modern age, ciphers have evloved dramatically. [AES](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard), [RSA](<https://en.wikipedia.org/wiki/RSA_(cryptosystem)>) and [Blowfish](<https://en.wikipedia.org/wiki/Blowfish_(cipher)>) are examples of ciphers that are integral part of contemporary encryption standards and systems.
```

File: files/en-us/glossary/cipher/index.md [hunk 5]
```diff
@@ -6,23 +6,24 @@ page-type: glossary-definition
...
+- {{Glossary("Symmetric-key cryptography", "symmetric key")}} algorithms use the same key to encode and decode a message. The key also must be sent securely if the message is to stay confidential.
+- {{Glossary("Public-key cryptography", "asymmetric key")}} algorithms use one key for encryption and the other for decryption.
 
 ## See also
 
-- [Cipher](https://en.wikipedia.org/wiki/Cipher) on Wikipedia
-- [Encryption and Decryption](/en-US/docs/Archive/Security/Encryption_and_Decryption)
+- [Web Crypto API](/en-US/docs/Web/API/Web_Crypto_API)
+- [SubtleCrypto supported algorithms](/en-US/docs/Web/API/SubtleCrypto#supported_algorithms)
 - [MDN Web Docs Glossary](/en-US/docs/Glossary)
```
Comment (PassionPenguin) [inline files/en-us/glossary/cipher/index.md:27]: wondering if this link could be remove, as it seems to be unnecessary ❓ 

(see discussion https://github.com/mdn/content/pull/33423#discussion_r1590843688)

```suggestion
- Glossary
```

File: files/en-us/glossary/public-key_cryptography/index.md [hunk 6]
```diff
@@ -15,10 +15,13 @@ When used for encryption, the public key is used to encrypt and the private key
 As a result, it's common for an encryption system to use a symmetric algorithm to encrypt the message, then a public-key system to encrypt the symmetric key. This arrangement can confer the benefits of both systems.
 
 Commonly used public-key cryptosystems are RSA (for both signing and encryption), DSA (for signing) and Diffie-Hellman (for key agreement).
+Elliptic-curve cryptography offers an alternative approach to these cryptographic systems with the benefit of using comparatively smaller key sizes, faster operations, and smaller storage and transmission.
+Popular elliptic-curve algorithms include [Elliptic-curve Diffie–Hellman](https://en.wikipedia.org/wiki/Elliptic-curve_Diffie%E2%80%93Hellman) and [Elliptic Curve Digital Signature Algorithm](https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm), which developers will often encounter abbreviated as **ECDH** and **ECDSA**.
```
Comment (PassionPenguin) [inline files/en-us/glossary/public-key_cryptography/index.md:19]: `<https://en.wikipedia.org/wiki/Elliptic-curve_Diffie–Hellman>` for better readability and maintainability :)

(is it necessary to add braces `<` & `>` ❔)

```suggestion
Popular elliptic-curve algorithms include [Elliptic-curve Diffie–Hellman](<https://en.wikipedia.org/wiki/Elliptic-curve_Diffie–Hellman>) and [Elliptic Curve Digital Signature Algorithm](https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm), which developers will often encounter abbreviated as **ECDH** and **ECDSA**.
```

File: files/en-us/glossary/public-key_cryptography/index.md [hunk 7]
```diff
@@ -15,10 +15,13 @@ When used for encryption, the public key is used to encrypt and the private key
...
 
 ## See also
 
 - [Web Crypto API](/en-US/docs/Web/API/Web_Crypto_API)
+- [SubtleCrypto supported algorithms](/en-US/docs/Web/API/SubtleCrypto#supported_algorithms)
 - [MDN Web Docs Glossary](/en-US/docs/Glossary)
```
Comment (PassionPenguin) [inline files/en-us/glossary/public-key_cryptography/index.md:25]: wondering if this link could be remove, as it seems to be unnecessary ❓ 

(see discussion https://github.com/mdn/content/pull/33423#discussion_r1590843688)

```suggestion
- Glossary
```

File: files/en-us/glossary/public-key_cryptography/index.md [hunk 8]
```diff
@@ -15,10 +15,13 @@ When used for encryption, the public key is used to encrypt and the private key
...
 
   - {{Glossary("Symmetric-key cryptography")}}
```
Comment (PassionPenguin) [inline files/en-us/glossary/public-key_cryptography/index.md:27]: maybe we can add a link to the wikipedia about the term?

```suggestion
  - {{Glossary("Symmetric-key cryptography")}}
  
- [Public-key cryptography](https://en.wikipedia.org/wiki/Public-key_cryptography) on Wikipedia
```

File: files/en-us/glossary/symmetric-key_cryptography/index.md [hunk 9]
```diff
@@ -12,13 +12,14 @@ This is usually contrasted with {{Glossary("public-key cryptography")}}, in whic
...
+Most symmetric-key algorithms currently in use are block {{Glossary("Cipher", "ciphers")}}: this means that they encrypt data one block at a time. The size of each block is fixed and determined by the algorithm: for example AES uses 16-byte blocks. Block ciphers are always used with a _{{Glossary("Block cipher mode of operation", "mode")}}_, which specifies how to securely encrypt messages that are longer than the block size. For example, AES is a cipher, while CTR, CBC, and GCM are all modes. Using an inappropriate mode, or using a mode incorrectly, can completely undermine the security provided by the underlying cipher.
 
 ## See also
 
+- [Web Crypto API](/en-US/docs/Web/API/Web_Crypto_API)
+- [SubtleCrypto supported algorithms](/en-US/docs/Web/API/SubtleCrypto#supported_algorithms)
 - [MDN Web Docs Glossary](/en-US/docs/Glossary)
```
Comment (PassionPenguin) [inline files/en-us/glossary/symmetric-key_cryptography/index.md:21]:
```suggestion
- Glossary
```

**Review by PassionPenguin — COMMENTED** [review]
that’s perfect improvements! here are some suggestions that may make it even better 🙇‍♂️ 

(some are improvements on the text of untouched parts, but since it’s a improvement, should we cover them up!

File: files/en-us/glossary/cipher/index.md (hunk 4 above)
Comment (bsmth) [reply files/en-us/glossary/cipher/index.md:12]: I'm fine with that one 👍🏻

File: files/en-us/glossary/public-key_cryptography/index.md (hunk 6 above)
Comment (bsmth) [reply files/en-us/glossary/public-key_cryptography/index.md:19]: better, tnx. we can remove the angle brackets, in that case.

**Comment by PassionPenguin** [comment]
@bsmth thx for that! bravo!

**Review by pepelsbey — APPROVED** [review]

**Comment by bsmth** [comment]
Thank you!

**Pull request merged**